keyboards/nullbitsco/snap/keymaps/bongo_reactive/keymap.c

```diff
@@ -0,0 +1,133 @@
+    KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,          KC_NO,    KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,
+    KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,          KC_NO,    KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,          KC_NO,
+    KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,    KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,          KC_NO,  KC_NO,
+    KC_NO,  KC_NO,  KC_NO,  KC_NO,          KC_NO,          KC_NO,    KC_NO,                  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO
+  ),
+    [_VIA2] = LAYOUT_all(
+            KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,    KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,
+    KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,    KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,
+    KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,          KC_NO,    KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,
+    KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,          KC_NO,    KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,          KC_NO,
+    KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,    KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,          KC_NO,  KC_NO,
+    KC_NO,  KC_NO,  KC_NO,  KC_NO,          KC_NO,          KC_NO,    KC_NO,                  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO
+  )
+};
+// clang-format on
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
+    [_BASE] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_MPRV, KC_MNXT) },
+    [_VIA1] = { ENCODER_CCW_CW(KC_NO, KC_NO),     ENCODER_CCW_CW(KC_NO, KC_NO) },
+    [_VIA2] = { ENCODER_CCW_CW(KC_NO, KC_NO),     ENCODER_CCW_CW(KC_NO, KC_NO) }
+};
+#endif
+
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+    if (is_keyboard_left())
+        return OLED_ROTATION_0;
+    else
+        return OLED_ROTATION_180;
+}
+
+static void render_status(void) {
+    oled_set_cursor(0, 0);
+    oled_write_P(PSTR("SNAP75 "), false);
+    oled_write_P(PSTR("Layer "), false);
+    switch (get_highest_layer(layer_state)) {
+        case _VIA1:
+            oled_write_P(PSTR("FN1 "), false);
+            break;
+        case _VIA2:
+            oled_write_P(PSTR("FN2 "), false);
+            break;
+        default: // use BASE case as default
+            oled_write_P(PSTR("Base"), false);
+    }
+
+    // Host Keyboard LED Status
+    oled_set_cursor(0, 1);
+    static uint8_t persistent_led_state = 0;
+    uint8_t led_usb_state = host_keyboard_leds();
+
+    // Only update if the LED state has changed
+    // Otherwise, the OLED will not turn off if an LED is on.
+    if (persistent_led_state != led_usb_state) {
+        persistent_led_state = led_usb_state;
+        
+        oled_write_ln_P(PSTR(""), false);
+
+        if (IS_LED_ON(led_usb_state, USB_LED_CAPS_LOCK)) {
+            oled_set_cursor(0, 1);
```

Comment:
just a heads up, it's best to avoid `oled_set_cursor`, as it can unintentionally force refreshes, which can take more time and keep the screen on longer than it should

Reply:
Thanks, that's good to know. Any recommendations on how to avoid it? I had to cache the previous LED state here so that the LED update didn't write to the OLED every cycle, which sounds exactly like what you're describing.

Reply:
Not really.  The biggest thing is just try to render stuff in order.  also `oled_advance_page(true)` will move to the next line (and fill in the in between with spaces/blanks).

But in my experience, a lot of trial and error is what will get you there.